senior_game_client/src/http.rs

```diff
@@ -0,0 +1,154 @@
+use std::sync::{Arc, Mutex};
+use std::thread;
+
+use bevy::prelude::*;
+#[cfg(test)]
+use mockito;
+use reqwest::blocking::{Client, RequestBuilder, Response};
+use reqwest::{Error, StatusCode};
+use serde_json::{Map, Value};
+use url::Url;
+
+#[allow(dead_code)]
+pub enum WebRequestVerb {
+  Get,
+  Post,
+}
+
+pub const LOGIN_SUFFIX: &str = "/session";
+
+pub fn domain_url() -> Url {
+  #[cfg(not(test))]
+  let domain: &str = "https://accounts.senior.realliance.net";
+
+  #[cfg(test)]
+  let domain: &str = &mockito::server_url();
+
+  Url::parse(domain).unwrap()
+}
+
+pub fn login_route() -> String {
+  domain_url().join(LOGIN_SUFFIX).unwrap().into_string()
+}
+
+pub struct LoginRequestTag;
+
+pub struct HttpRequest {
+  pub verb: WebRequestVerb,
+  pub url: String,
+  pub body: Option<Value>,
+}
+
+#[derive(Default)]
+pub struct HttpInProgress(Arc<Mutex<Option<Result<Response, Error>>>>);
+
+#[derive(PartialEq, Debug)]
+pub struct HttpResponse {
+  pub is_error: bool,
+  pub status: Option<StatusCode>,
+  pub response_body: Option<Value>,
+}
+
+impl HttpResponse {
+  pub fn get_json_object(&self) -> &Map<String, Value> {
+    return self.response_body.as_ref().unwrap().as_object().unwrap();
+  }
+
+  pub fn get_value(&self, field: &str) -> String {
+    return self.get_json_object().get(field).unwrap().as_str().unwrap().to_string();
+  }
+}
```

Comment:
This code is so horribly inefficient and unsafe that I lost braincells reading it